EventListener/ReportSubscriber.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+namespace MauticPlugin\LeuchtfeuerCompanySegmentsBundle\EventListener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\Expression\CompositeExpression;
+use Mautic\CoreBundle\Helper\InputHelper;
+use Mautic\LeadBundle\Model\CompanyReportData;
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+use Mautic\ReportBundle\ReportEvents;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegmentRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ReportSubscriber implements EventSubscriberInterface
+{
+    public const CONTEXT_COMPANY_SEGMENTS         = 'company_segments';
+    public const COMPANY_TABLE                    = 'companies';
```

Comment:
Please use existing constants for tables.

Reply:
Is it really necessary? The context is not necessary to be called exactly as the table. It could also be  comp_seg and there wouldn't be a problem

Reply:
As far as i can see this is the name of the table you are querying from.